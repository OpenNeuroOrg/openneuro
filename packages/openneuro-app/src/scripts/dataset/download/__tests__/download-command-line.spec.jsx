import React from "react"
import { render, screen } from "@testing-library/react"
import { DownloadSampleCommand } from "../download-command-line.jsx"

const defProps = { datasetId: "ds000001" }

describe("dataset/download", () => {
  describe("DownloadSampleCommand component", () => {
    it("renders successfully", () => {
      const { asFragment } = render(<DownloadSampleCommand {...defProps} />)
      expect(asFragment()).toMatchSnapshot()
    })
    it("drafts show draft flag", () => {
      render(<DownloadSampleCommand {...defProps} />)
      expect(screen.getByRole("figure")).toHaveTextContent("--draft")
      expect(screen.queryByText("--version")).not.toBeInTheDocument()
    })
    it("snapshots show snapshot flag", () => {
      render(<DownloadSampleCommand {...defProps} snapshotTag="1.0.0" />)
      expect(screen.getByRole("figure")).toHaveTextContent("--version")
      expect(screen.queryByText("--draft")).not.toBeInTheDocument()
    })
  })
})
