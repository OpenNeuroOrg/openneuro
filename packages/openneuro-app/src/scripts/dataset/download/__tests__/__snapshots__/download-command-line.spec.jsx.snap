// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dataset/download > DownloadSampleCommand component > renders successfully 1`] = `
<DocumentFragment>
  <pre
    class="css-xak41w"
    role="figure"
  >
    # Login by following the prompts
    <br />
    deno run -A jsr:@openneuro/cli login
    <br />
    # Download the repository
    <br />
    deno run -A jsr:@openneuro/cli download --draft ds000001 ds000001-download/
  </pre>
</DocumentFragment>
`;
